src/pages/SupplierRequisitionPage.js

```diff
@@ -1,666 +1,334 @@
-            }}
-          />
-        );
-      case VIEW_REGIMEN_DATA:
-        return (
-          <RequisitionRegimenModalTable
-            database={database}
-            requisition={requisition}
-            genericTablePageStyles={genericTablePageStyles}
-          />
-        );
+        return null;
     }
-  };
-
-  renderButtons = () => {
-    const { requisition } = this.props;
-    const { program, thresholdMOS, parsedCustomData } = requisition;
-    const hasRegimenData =
-      parsedCustomData && parsedCustomData.regimenData && parsedCustomData.regimenData.length !== 0;
-    const UseSuggestedQuantitiesButton = () => (
-      <PageButton
-        style={{
-          ...globalStyles.topButton,
-          ...(program ? { marginLeft: 5 } : {}),
-        }}
-        text={buttonStrings.use_suggested_quantities}
-        onPress={this.onUseSuggestedQuantities}
-        isDisabled={requisition.isFinalised}
-      />
-    );
-
-    const ViewRegimenDataButton = () => (
-      <PageButton
-        style={{
-          ...globalStyles.topButton,
-          ...(program ? { marginLeft: 5 } : {}),
-        }}
-        text={buttonStrings.view_regimen_data}
-        onPress={() => this.openModal(MODAL_KEYS.VIEW_REGIMEN_DATA)}
-      />
-    );
-
-    const ThresholdMOSToggle = () => {
-      const { useThresholdMOS } = this.state;
-      const onPress = () => this.setState({ useThresholdMOS: !useThresholdMOS }, this.refreshData);
-      const toggleProps = [
-        {
-          text: programStrings.hide_over_stocked,
-          isOn: useThresholdMOS,
-          onPress,
-        },
-        {
-          text: programStrings.show_over_stocked,
-          isOn: !useThresholdMOS,
-          onPress,
-        },
-      ];
-      return <ToggleBar style={globalStyles.toggleBar} toggles={toggleProps} />;
-    };
+  });
+
+  const renderRow = useCallback(
+    listItem => {
+      const { item, index } = listItem;
+      const rowKey = keyExtractor(item);
+      const { row, alternateRow } = newDataTableStyles;
+      return (
+        <DataTableRow
+          rowData={data[index]}
+          rowState={dataState.get(rowKey)}
+          rowKey={rowKey}
+          style={index % 2 === 0 ? alternateRow : row}
+          columns={columns}
+          isFinalised={isFinalised}
+          dispatch={dispatch}
+          focusCellAction={focusCell}
+          focusNextAction={focusNext}
+          getAction={getAction}
+        />
+      );
+    },
+    [data, dataState]
+  );
 
-    const CreateAutomaticOrderButton = () => (
+  const AddMasterListItemsButton = useCallback(
+    () => (
       <PageButton
         style={globalStyles.leftButton}
-        text={buttonStrings.create_automatic_order}
-        onPress={this.onCreateAutomaticOrder}
-        isDisabled={requisition.isFinalised}
+        text={buttonStrings.add_master_list_items}
+        onPress={() => runWithLoadingIndicator(() => dispatch(addMasterListItems('Requisition')))}
```

Comment:
two anonymous functions to create in a render!
not a big problem, was wondering if that could be avoided though. Also `runWithLoadingIndicator` is used to wrap the method passed to `onPress` - is it possible to have the `PageButton` component handle that, and just pass in the on press function, along with a boolean prop indicating that loading indicator should be used?
Or have another component that almost replicates `PageButton` but with loading indication? `PageButtonWithLoadingIndicator` ( just demonstrating there that naming is hard and it is much better to have a prop on `PageButton` )

Reply:
I like the idea of `PageButtonWithLoadingIndicator` - especially since calling this method re-renders every page in the navigation stack, currently!